icaruscode/CRT/CRTUtils/CRTDetSimAlg.cc

```diff
@@ -496,7 +496,9 @@ namespace icarus{
 	      path += "/";
 	    }
 	  }
-	  if (path.find(adGeo.Name())!=std::string::npos && path.find("strip"+stripn)!=std::string::npos) break;
+	  if ( (path.find(adGeo.Name())!=std::string::npos && path.find("strip"+stripn)!=std::string::npos) || //refactored
+	       (std::string(adsGeo.TotalVolume()->GetName()).find("_")!=std::string::npos && path.find(adsGeo.TotalVolume()->GetName())!=std::string::npos) ) //legacy
```

Comment:
Isn't whether the volume name contains an underscore known since the beginning?
I think's is faster and more readable to cache that information:
```cpp
  // whether we are using the "refactored" geometry naming scheme
  bool const isRefactored = std::string(adsGeo.TotalVolume()->GetName()).find("_") != std::string::npos;
```
```suggestion
	       (isRefactored && path.find(adsGeo.TotalVolume()->GetName())!=std::string::npos) ) //legacy
```

Reply:
Thanks for the suggestion. I assume the variable should be called "isLegacy" rather than "isRefactored" thoughj

Reply:
Possible. I am lost with respect to which has `"_"` and which does not.

Reply:
Legacy has "_"